src/primitives/block.cpp

```diff
@@ -48,9 +48,19 @@ uint256 CBlockHeader::GetX16RTPoWHash(bool fSetVeilDataHashNull) const
 uint256 CBlockHeader::GetSha256DPoWHash() const
 {
     CSha256dDataInput input(*this);
-
     uint256 dataHash = SerializeHash(input);
+    CSha256dInput sha256Final(*this, dataHash);
+    return SerializeHash(sha256Final);
+}
+
+uint256 CBlockHeader::GetDataHash() const
+{
+    CSha256dDataInput input(*this);
```

Comment:
Since this is specific to Sha256; change the `GetDataHash()` to be reflective of that

Reply:
`uint256 CBlockHeader::GetSha256dMidstate()` ?

Reply:
Sure; that works. Use your discretion; but just as long as it is identifiable as a sha256 function [I figured `GetSha256dDataHash()`, but whatever makes sense I'm good with.

Reply:
amended in 5650b36